api/prove.js

```diff
@@ -19,98 +17,124 @@ import {
  * @param {boolean} enableLog - enable logging or not
  * @returns {object} - proof details in json
  */
-export async function prove(wasmUnit8Array, privateInputStr, publicInputStr, zkwasmProverUrl, userPrivateKey, enableLog=true) {
-    let result = {"instances":null, "batch_instances":null, "proof":null, "aux":null, "md5": null, "taskId": null};
+export async function prove(
+  wasmUnit8Array,
+  privateInputStr,
+  publicInputStr,
+  zkwasmProverUrl,
+  userPrivateKey,
+  enableLog = true
+) {
+  let result = {
+    instances: null,
+    batch_instances: null,
+    proof: null,
+    aux: null,
+    md5: null,
+    taskId: null,
+  };
+
+  // Prove mode
+  const privateInputArray = privateInputStr.trim().split(" ");
+  const publicInputArray = publicInputStr.trim().split(" ");
+
+  // Message and form data
+  const md5 = ZkWasmUtil.convertToMd5(wasmUnit8Array).toUpperCase();
+
+  result["md5"] = md5;
+
+  let [response, isSetUpSuccess, errorMessage] = await zkwasm_prove(
+    zkwasmProverUrl,
+    userPrivateKey,
+    md5,
+    publicInputArray,
+    privateInputArray
+  ).catch((error) => {
+    throw error;
+  });
+
+  if (enableLog) {
+    console.log(`[*] IMAGE MD5: ${md5}`, "\n");
+  }
+
+  if (isSetUpSuccess) {
+    //   console.log(`[+] IMAGE MD5: ${response.data.result.md5}`, "\n");
+
+    const taskId = response.data.result.id;
```

Comment:
Will need to also return taskId in result, so that user of this method can access it in prove's return value, then call waitProve with that value.